searchcore/src/vespa/searchcore/proton/matchengine/matchengine.cpp

```diff
@@ -113,18 +113,21 @@ void
 MatchEngine::performSearch(search::engine::SearchRequest::Source req,
                            search::engine::SearchClient &client)
 {
-   auto ret = std::make_unique<search::engine::SearchReply>();
+    auto ret = std::make_unique<search::engine::SearchReply>();
 
-    if (req.get()) {
+    const search::engine::SearchRequest * searchRequest = req.get();
+    if (searchRequest) {
+        // 3 is the minimum level required for backend tracing.
+        searchRequest->setTraceLevel(search::fef::indexproperties::trace::Level::lookup(searchRequest->propertiesMap.modelOverrides()), 3);
```

Comment:
setTraceLevel in the search request should just set the trace level, not initiate tracing.
start(3) should be separate and called from here. (matched with done() call below).

doing it this way will break proto search request conversion since we can no longer use setTraceLevel/getTraceLevel for trace level (which would make sense now that trace level is a concept on the request itself), but need to insert it into the appropriate key/value map instead.